test/spec_server.rb

```diff
@@ -334,6 +334,28 @@ def start(*) [self.class, :started] end
     c.start.must_equal [c, :started]
   end
 
+  def start_server(server)
+    begin
+      ready = Thread::Queue.new
+      thread = Thread.new do
+        begin
+          server.start { |s| ready.push(true) }
+        ensure
+          ready.close
+        end
+      end
+
+      unless ready.pop
+        thread.join
```

Comment:
This seems very hard to follow, what is it trying to do?
Can't it just always join?

Reply:
It only joins if it fails to start the server.

Reply:
Another way would be to set `abort_on_exception` for that thread as the first thing it does. Then no need for the queue.